modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParserUtils.java

```diff
@@ -213,6 +222,66 @@ private static void checkAnnotationsNotUsedMoreThanOnce(List<? extends Extension
     return connectionParameter;
   }
 
+  public static Optional<DeprecationModel> getDeprecationModel(ExtensionParameter extensionParameter) {
```

Comment:
Not saying that we need to do this now, but If we start adding all the parsing utils here maybe it will get a bit crowded. In the future maybe we create a util package here.